platform/feature-api/api/src/pod-api/tests/uri.test.ts

```diff
@@ -0,0 +1,24 @@
+import { createUUID, isPolypodUri, PolyUri } from "../uri";
+
+describe("createUUID tests", () => {
+    it("should generate an UUID in the required format", () => {
+        expect(createUUID()).toMatch(/^\w{8}-\w{4}-4\w{3}-\w{4}-\w{12}$/);
+    });
+
+    it("should generate different subsequent UUIDs", () => {
+        let lastUUID = createUUID();
+        for (let i = 0; i < 100; i++) {
+            const thisUUID = createUUID();
+            expect(thisUUID).not.toEqual(lastUUID);
+            lastUUID = thisUUID;
+        }
+    });
+});
+
+describe("isPolypodUri()", () => {
+    it("recognizes good and bad URIs", () => {
+        const polyUri: PolyUri = new PolyUri();
+        expect(isPolypodUri(polyUri.toString())).toBe(true);
+        expect(isPolypodUri("foobargaz")).toBe(false);
```

Comment:
nit: (leaving this here as a future reference for unit tests...)

the newlines play quite big important in order to point out the 'arrange-act-assert' pattern

```suggestion
       // arrange
        const polyUri = new PolyUri();
        
        // act
        const actualTrue = isPolypodUri(polyUri.toString());
        const actualFalse = isPolypodUri("foobargaz")
        
        // assert
        expect(actualTrue).toBe(true);
        expect(actualFalse).toBe(false);
```
Meaning that the Arrange section we only have code required to setup that specific test. 
Here objects would be created, mocks setup (if we are using one) and potentially expectations would be set. 
Then there is the Act, which should be the invocation of the method being tested. 
And on Assert we would simply check whether the expectations were met. 

FYI: https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/
(example in python, but methodology is the same)

Reply:
In that case, I would move the creation to a real Arrange, or beforeeach, section. I can't apply this now, anyway, so I'll do it that way.

Reply:
This can't be applied since I have used your other suggestion. I guess this can be resolved, right?